applications/postgres.md

```diff
@@ -1,13 +1,215 @@
+
+PostgreSQL has earned a strong reputation for its reliability, data integrity, and correctness. 
+
+*	It runs on all major operating systems, including Linux, UNIX (AIX, BSD, HP-UX, SGI IRIX, MacOS, Solaris, Tru64), and Windows. 
+
+*	It is fully ACID compliant, has full support for foreign keys, joins, views, triggers, and stored procedures (in multiple languages). 
+
+*	It includes most SQL:2008 data types, including INTEGER, NUMERIC, BOOLEAN, CHAR, VARCHAR, DATE, INTERVAL, and TIMESTAMP. 
+    
+*	It has native programming interfaces for C/C++, Java, .Net, Perl, Python, Ruby, Tcl, ODBC, among others, and exceptional documentation.
+
+PostgreSQL on Portworx and kubernetes helped to utilize resources even much better than virtual machines and also provide isolation from other apps which are deployed on the same machine.
+
+### Prerequisites
+
+*	3-node Portworx cluster running on Kubernetes.
+
+### Step 1 - Create PostgreSQL Portworx StorageClass
+Create a file name called `px-postgres-sc.yaml` for Portworx StorageClass for PostgreSQL.
+
+```
+
+##### Portworx storage class
+kind: StorageClass
+apiVersion: storage.k8s.io/v1beta1
+metadata:
+    name: px-postgres-sc
+provisioner: kubernetes.io/portworx-volume
+parameters:
+   repl: "2"
+
+```
+`kubectl apply -f px-postgres-sc.yaml`
+
+### Step 2 - Create PostgreSQL Portworx PersistentVolumeClaim
+Create a file name called `px-postgres-vol.yaml` for Portworx PersistentVolumeClaim for PostgreSQL.
+
+```
+
+##### Portworx persistent volume claim
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+   name: postgres-data
+   annotations:
+     volume.beta.kubernetes.io/storage-class: px-postgres-sc
+spec:
+   accessModes:
+     - ReadWriteOnce
+   resources:
+     requests:
+       storage: 1Gi
+
+```
+`kubectl apply -f px-postgres-vol.yaml`
+
+### Step 3 - Deploy PostgreSQL using Kubernetes Deployment
+ 
+Deploying PostgreSQL on Kubernetes have following prerequisites. 
+Create a file name `px-postgres-app.yaml` for PostgreSQL. 
+ 
+We need to Define the Environment Variables for PostgreSQL
+
+1. POSTGRES_USER (Super Username for PostgreSQL)
+
+2. POSTGRES_PASSWORD (Super User password for PostgreSQL)
+
+3. PGDATA (Data Directory for PostgreSQL Database)
+
+```
+
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: postgres
+spec:
+  strategy:
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 1
+    type: RollingUpdate
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: postgres
+    spec:
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: px/enabled
+                operator: NotIn
+                values:
+                - "false"
+      containers:
```

Comment:
Update the spec for using stork as the scheduler.

Stork is documented here: https://github.com/libopenstorage/stork